### PR TITLE
OAK-9679 support testing with DynamoDBLocal on Apple Silicon

### PR DESCRIPTION
**Pull request by kwin**

### CODE REVIEW

File: oak-segment-aws/pom.xml [hunk 1]
```diff
@@ -309,6 +308,14 @@
             <type>dll</type>
             <scope>test</scope>
         </dependency>
+        <!-- add native implementation for Apple silicon from fork -->
+        <dependency>
+            <groupId>io.github.ganadist.sqlite4java</groupId>
```
Comment (thomasmueller) [inline oak-segment-aws/pom.xml:314]: according to https://github.com/ganadist/sqlite4java
"sqlite4java is licensed under Apache License 2.0."

File: oak-segment-aws/pom.xml [hunk 2]
```diff
@@ -120,7 +120,6 @@
             <plugin>
                 <groupId>org.apache.maven.plugins</groupId>
                 <artifactId>maven-dependency-plugin</artifactId>
-                <version>3.1.1</version>
```
Comment (thomasmueller) [inline oak-segment-aws/pom.xml]: How is this change related?

File: oak-segment-aws/pom.xml [hunk 3]
```diff
@@ -309,6 +308,14 @@
             <type>dll</type>
             <scope>test</scope>
         </dependency>
+        <!-- add native implementation for Apple silicon from fork -->
+        <dependency>
+            <groupId>io.github.ganadist.sqlite4java</groupId>
+            <artifactId>libsqlite4java-osx-aarch64</artifactId>
+            <version>${sqlite4java.version}</version>
+            <type>dylib</type>
+            <scope>test</scope>
```
Comment (thomasmueller) [inline oak-segment-aws/pom.xml:318]: There is no test yet, just adding the dependency... what about we first add a test case before we add a dependency (in the same PR)?

File: oak-segment-aws/pom.xml (hunk 1 above)
Comment (kwin) [reply oak-segment-aws/pom.xml:314]: sqlite4java was always a dependency, the fork uses the same license, so this should be fine.

File: oak-segment-aws/pom.xml (hunk 2 above)
Comment (kwin) [reply oak-segment-aws/pom.xml]: not related, gonna remove

File: oak-segment-aws/pom.xml (hunk 3 above)
Comment (kwin) [reply oak-segment-aws/pom.xml:318]: This is not true, almost all existing tests fail on Apple M1 without this dependency, they succeed with that. Unfortunately neither GitHub Actions nor ASF Jenkins supports (testing with) that platform.

Comment (kwin) [reply oak-segment-aws/pom.xml:318]: Also notice that this is only a test dependency!

**Review by mreutegg — APPROVED** [review]
Looks good to me. Tests still succeed on my machine with Intel CPU.

@thomasmueller does this work for you? I remember you are also on an Apple M1.
